### PR TITLE
Update Intel/impi version to Intel 2021/2022

### LINKED ISSUE #1113: Update intel/impi compiler to 2022
## Description
Intel compiler versions 2019 and 2018 will not be supported on hera/orion. The ufs-weather-model will be updated to use new intel 2022 version.
 
## Solution
Add a clear and concise description of what you want to happen.

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- https://github.com/NOAA-EMC/hpc-stack/issues/378

### PR DESCRIPTION
**Pull request by junwang-noaa**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

In this PR, the Intel compiler will be updated to Intel 2021/2022 for hera, orion, jet, gaea, and Cheyenne. The wcoss dell/cray will go down in two months and additional disk space cleanup is required for installing the hpc-stack with the new Intel compiler, so the intel compiler is not updated on these two platforms. 

Additional changes are: 
1) The environment variable I_MPI_DAPL_UD is turned off and all the GOCART tests passed.
2) png library is changed to libpng in ufs_common/ufs_common_debug.

The branch was tested on hera, orion, jet and gaea. Butterfly tests were conducted on orion/hera. The results are summarized in the issue #1113.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes https://github.com/ufs-community/ufs-weather-model/issues/1113

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI: https://github.com/ufs-community/ufs-weather-model/actions/runs/2358569553

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

File: modulefiles/ufs_common [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module load jasper/2.0.22
```
Comment (DusanJovic-NOAA) [inline modulefiles/ufs_common]: Should we update jasper to 2.0.25? And maybe g2.

See: https://github.com/NOAA-EMC/hpc-stack/issues/432#issuecomment-1098352185

File: modulefiles/ufs_gaea.intel [hunk 2]
```diff
@@ -25,12 +25,12 @@ module load gcc/8.3.0
...
 
 module load hpc/1.1.0
-module load hpc-intel/18.0.6.288
+module load hpc-intel/2021.3.0
 module load hpc-cray-mpich/7.7.11
 
 module load ufs_common
 
-setenv I_MPI_DAPL_UD 1
+#setenv I_MPI_DAPL_UD 1
```
Comment (DusanJovic-NOAA) [inline modulefiles/ufs_gaea.intel]: If this is not needed anymore I would delete this line

Comment (junwang-noaa) [reply modulefiles/ufs_gaea.intel]: removed.

File: modulefiles/ufs_common (hunk 1 above)
Comment (junwang-noaa) [reply modulefiles/ufs_common]: Thanks for reminding me those libraries. I am testing the code with g2 (3.4.5), Jasper (2.0.25), PIO (2.5.3), libpng (1.6.37).

Comment (junwang-noaa) [reply modulefiles/ufs_common]: The libraries are tested on hera, it is confirmed that these updated libraries do not change results.

**Comment by jkbk2004** [comment]
@junwang-noaa I am testing cpld_control_p8 on cheyenne. It seems results changes. Maybe I need butterfly test for validation.

**Comment by junwang-noaa** [comment]
@jkbk2004 Updating Intel/impi version will change the results. But the library updates (g2 (3.4.5), Jasper (2.0.25), PIO (2.5.3), libpng (1.6.37)) will not change results. Please let me know from which libraries that you found result change.

File: modulefiles/ufs_cheyenne.intel [hunk 3]
```diff
@@ -11,16 +11,16 @@ module load python/3.7.9
...
 module load ncarenv/1.3
-module load intel/2021.2
-module load mpt/2.22
+module load intel/2022.1
+module load mpt/2.25
 module load ncarcompilers/0.5.0
 module unload netcdf
 
-module use /glade/work/epicufsrt/GMTB/tools/hpc-stack-v1.2.0_6eb6/modulefiles/stack
+module use /glade/work/epicufsrt/GMTB/tools/intel/2022.1/hpc-stack-v1.2.0_6eb6/modulefiles/stack
```
Comment (DusanJovic-NOAA) [inline modulefiles/ufs_cheyenne.intel:19]: hpc stack for gnu is located under `hpc-stack-v1.2.0` while intel is under `hpc-stack-v1.2.0_6eb6`. Why `_6eb6` suffix? Is it too late to change this?

**Review by DusanJovic-NOAA — APPROVED** [review]

File: modulefiles/ufs_cheyenne.intel (hunk 3 above)
Comment (jkbk2004) [reply modulefiles/ufs_cheyenne.intel:19]: @DusanJovic-NOAA hpc-stack develop branch has an issue with intel/mpt on Cheyenne. So, I still have to use hash (6eb6) for intel. But gnu has no problem with hpc-stack develop branch. So I have to separate the gnu and intel locations.

Comment (DusanJovic-NOAA) [reply modulefiles/ufs_cheyenne.intel:19]: Ok. Let's hope the issue is resolved soon.

Comment (jkbk2004) [reply modulefiles/ufs_cheyenne.intel:19]: @DusanJovic-NOAA Yes, I think we keep moving on ufs-wm side as is. When we try to get away from mpt on cheyenne, the issue can be resolved by itself, I guess.

Comment (climbfuji) [reply modulefiles/ufs_cheyenne.intel:19]: `mpt` doesn't support `mpi_f08`, thus when @DusanJovic-NOAA's PR for that goes in we have to switch anyway

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/937515199/20220519204511/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/glade/scratch/epicufsrt/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220517/GNU
Please make changes and add the following label back: cheyenne-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
[BL] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/937515199/20220520024510/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/lfs4/HFIP/h-nems/emc.nemspara/RT/NEMSfv3gfs//develop-20220517/INTEL
Please make changes and add the following label back: jet-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
[BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/937515199/20220519214520/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220517/INTEL
Please make changes and add the following label back: orion-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/937515199/20220519204620/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/glade/scratch/epicufsrt/GMTB/ufs-weather-model/RT/NEMSfv3gfs/develop-20220517/INTEL
Please make changes and add the following label back: cheyenne-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520024519/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220517/GNU
Please make changes and add the following label back: hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
[BL] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/937515199/20220520024506/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/lustre/f2/pdata/ncep_shared/emc.nemspara/RT/NEMSfv3gfs/develop-20220517/INTEL
Please make changes and add the following label back: gaea-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520024733/ufs-weather-model
[BL] ERROR: Baseline location exists before creation:
/scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220517/INTEL
Please make changes and add the following label back: hera-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520033008/ufs-weather-model
Please make changes and add the following label back: hera-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
[BL] Repo location: /glade/scratch/epicufsrt/autort/tests/auto/pr/937515199/20220519214511/ufs-weather-model
[BL] Error: Test cpld_debug_p8 006 failed in run_test failed 
[BL] Error: Test control_debug 044 failed in run_test failed 
[BL] Error: Test control_CubedSphereGrid_debug 045 failed in run_test failed 
[BL] Error: Test control_wrtGauss_netcdf_parallel_debug 046 failed in run_test failed 
[BL] Error: Test control_stochy_debug 047 failed in run_test failed 
[BL] Error: Test control_lndp_debug 048 failed in run_test failed 
[BL] Error: Test control_csawmg_debug 049 failed in run_test failed 
[BL] Error: Test control_csawmgt_debug 050 failed in run_test failed 
[BL] Error: Test control_ras_debug 051 failed in run_test failed 
[BL] Error: Test control_diag_debug 052 failed in run_test failed 
[BL] Error: Test control_debug_p8 053 failed in run_test failed 
[BL] Error: Test control_thompson_debug 054 failed in run_test failed 
[BL] Error: Test control_thompson_no_aero_debug 055 failed in run_test failed 
[BL] Error: Test control_thompson_extdiag_debug 056 failed in run_test failed 
[BL] Error: Test control_thompson_progcld_thompson_debug 057 failed in run_test failed 
[BL] Error: Test regional_debug 058 failed in run_test failed 
[BL] Error: Test rap_control_debug 059 failed in run_test failed 
[BL] Error: Test rap_diag_debug 060 failed in run_test failed 
[BL] Error: Test rap_cires_ugwp_debug 061 failed in run_test failed 
[BL] Error: Test rap_lndp_debug 062 failed in run_test failed 
[BL] Error: Test rap_flake_debug 063 failed in run_test failed 
[BL] Error: Test rap_progcld_thompson_debug 064 failed in run_test failed 
[BL] Error: Test rap_noah_debug 065 failed in run_test failed 
[BL] Error: Test rap_rrtmgp_debug 066 failed in run_test failed 
[BL] Error: Test rap_sfcdiff_debug 067 failed in run_test failed 
[BL] Error: Test rap_noah_sfcdiff_cires_ugwp_debug 068 failed in run_test failed 
[BL] Error: Test rrfs_v1beta_debug 069 failed in run_test failed 
[BL] Error: Test control_wam_debug 070 failed in run_test failed 
[BL] Error: Test datm_cdeps_debug_cfsr 098 failed in run_test failed 
[BL] Error: Test compile_001 failed in run_compile failed 
[BL] Error: Test compile_003 failed in run_compile failed 
[BL] Error: Test compile_004 failed in run_compile failed 
[BL] Error: Test compile_005 failed in run_compile failed 
[BL] Error: Test compile_006 failed in run_compile failed 
[BL] Error: Test compile_011 failed in run_compile failed 
[BL] Error: Test compile_012 failed in run_compile failed 
[BL] Error: Test compile_013 failed in run_compile failed 
[BL] Error: Test compile_015 failed in run_compile failed 
[BL] Error: Test compile_016 failed in run_compile failed 
Please make changes and add the following label back: cheyenne-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
[BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/937515199/20220519223019/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/937515199/20220519233519/ufs-weather-model
[RT] Error: Test compile_007 failed in run_compile failed 
Please make changes and add the following label back: orion-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520033635/ufs-weather-model
[BL] Error: Test control_c384gdas_wav 099 failed in run_test failed 
Please make changes and add the following label back: hera-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
[BL] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/937515199/20220520033007/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/937515199/20220520052255/ufs-weather-model
[RT] Error: Test rap_control 037 failed in run_test failed 
[RT] Error: Test rap_rrtmgp 038 failed in run_test failed 
Please make changes and add the following label back: jet-intel-BL

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA Corrected ufs_cheyenne.intel: from hpc-intel/2021.1 to hpc-intel/2022.2

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by BrianCurtis-NOAA** [comment]
I don't see hera.gnu and I don't see a label that would have started it after the previous failure. Is this being worked on?

**Comment by jkbk2004** [comment]
@MinsukJi-NOAA @BrianCurtis-NOAA do you want me to run RT on hera.gnu to confirm? so that we can merge.

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA @MinsukJi-NOAA hera.gnu should work since this pr has not much to do with gnu.

**Comment by BrianCurtis-NOAA** [comment]
@MinsukJi-NOAA Would you rather copy over the GNU baselines from the previous date, since i assume they won't change and add hera-gnu-RT, or just add the hera-gnu-BL?

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: RT
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520180014/ufs-weather-model
Please make changes and add the following label back: hera-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
> _**Automated RT Failure Notification**_ Machine: hera Compiler: gnu Job: RT [RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520180014/ufs-weather-model Please make changes and add the following label back: hera-gnu-RT

@MinsukJi-NOAA 
```
'cat: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/937515199/20220520180014/ufs-weather-model/tests/log_hera.gnu/compile_*_time.log: No such file or directory', "++ echo 'rt.sh error on line 858'", 'rt.sh error on line 858'
```

**Comment by BrianCurtis-NOAA** [comment]
```
Project: stmp1 
		Directory: /scratch2/NCEPDEV/stmp1 DiskInUse=420724 GB, Quota=400000 GB, Files=27611254, FileQUota=80000000
		Directory: /scratch1/NCEPDEV/stmp2 DiskInUse=418850 GB, Quota=400000 GB, Files=40878708, FileQUota=80000000
		Directory: /scratch2/NCEPDEV/stmp3 DiskInUse=420724 GB, Quota=400000 GB, Files=27611254, FileQUota=80000000
		Directory: /scratch1/NCEPDEV/stmp4 DiskInUse=418850 GB, Quota=400000 GB, Files=40878708, FileQUota=80000000
```
We're full, maybe thats it?

**Comment by MinsukJi-NOAA** [comment]
@jkbk2004 @BrianCurtis-NOAA  This is Hera.gnu compile error message.
```
Lmod has detected the following error: These module(s) or extension(s) exist
but cannot be loaded as requested: "libpng/1.6.37"
   Try: "module spider libpng/1.6.37" to see how to load the module(s).
```

**Comment by BrianCurtis-NOAA** [comment]
@MinsukJi-NOAA 

It loaded fine when i did it manually. I wonder why it had issue? Maybe something temp on Hera?

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA
> 
> It loaded fine when i did it manually. I wonder why it had issue? Maybe something temp on Hera?

@BrianCurtis-NOAA  I am not able to manually load libpng

```
[Minsuk.Ji@hecflow01 tests]$ module use ../modulefiles/
[Minsuk.Ji@hecflow01 tests]$ pwd
/scratch1/NCEPDEV/stmp4/Minsuk.Ji/PR1221/tests
[Minsuk.Ji@hecflow01 tests]$ module use /scratch1/NCEPDEV/nems/emc.nemspara/soft/modulefiles
[Minsuk.Ji@hecflow01 tests]$ module load miniconda3/3.7.3
[Minsuk.Ji@hecflow01 tests]$ 
[Minsuk.Ji@hecflow01 tests]$ module use /contrib/sutils/modulefiles
[Minsuk.Ji@hecflow01 tests]$ module load sutils
[Minsuk.Ji@hecflow01 tests]$ 
[Minsuk.Ji@hecflow01 tests]$ module load cmake/3.20.1
[Minsuk.Ji@hecflow01 tests]$ 
[Minsuk.Ji@hecflow01 tests]$ module use /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack
[Minsuk.Ji@hecflow01 tests]$ 
[Minsuk.Ji@hecflow01 tests]$ module load hpc/1.1.0
[Minsuk.Ji@hecflow01 tests]$ 
[Minsuk.Ji@hecflow01 tests]$ module load hpc-gnu/9.2.0
[Minsuk.Ji@hecflow01 tests]$ module load hpc-mpich/3.3.2
[Minsuk.Ji@hecflow01 tests]$ module load ufs_common
Lmod has detected the following error:  These module(s) or extension(s) exist but cannot be loaded as requested: "libpng/1.6.37"
   Try: "module spider libpng/1.6.37" to see how to load the module(s).
```

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA @MinsukJi-NOAA @junwang-noaa seems like no libpng with gnu there. we have to follow up with nceplibs team.
------------------ /scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/compiler/gnu/9.2.0 -------------------
   bacio/2.4.1        g2c/1.6.2                 grib_util/1.2.4   (D)    png/1.6.35             w3nco/2.4.1

**Comment by jkbk2004** [comment]
@BrianCurtis-NOAA @MinsukJi-NOAA I sent a message on hpc-stack slack and asked Kyle and Hang to install one.

**Comment by jkbk2004** [comment]
@MinsukJi-NOAA @BrianCurtis-NOAA I checked Hang just installed libpng/1.6.37. Its available on hera now. So nice to see his quick installation.

**Comment by MinsukJi-NOAA** [comment]
> @MinsukJi-NOAA @BrianCurtis-NOAA I checked Hang just installed libpng/1.6.37. Its available on hera now. So nice to see his quick installation.

Thanks so much @jkbk2004! hera.gnu RT is being rerun.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
[BL] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/937515199/20220520033006/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/937515199/20220520104410/ufs-weather-model
[RT] Error: Test cpld_restart_c384_p8 011 failed in run_test failed 
Please make changes and add the following label back: gaea-intel-BL

**Comment by junwang-noaa** [comment]
@MinsukJi-NOAA @jkbk2004 @BrianCurtis-NOAA Thank you very much for helping running RT.

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**